runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -12358,6 +12358,936 @@ static TR::Register *inlineIntrinsicIndexOf(TR::Node *node, TR::CodeGenerator *c
+    * The value in permuteVectorReg has not changed since it was generated so it doesn't need to be generated again.
+    */
+   generateTrg1Src2Instruction(cg, vectorCompareOp, node, vec2Reg, searchVectorReg, targetVectorReg);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::beq, node, notFoundLabel, cr6Reg);
+   generateLabelInstruction(cg, TR::InstOpCode::b, node, foundEndVectorRetryLabel);
+
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, notSmallLabel);
+
+   /*
+    * Check if we already have 16-byte alignment. Vector loads require 16-byte alignment, so if we
+    * aren't properly aligned, we'll need to handle comparisons specially until we achieve 16-byte
+    * alignment. At this point, tempReg is equal to currentAddressReg with the lowest 4 bits clear.
+    */
+   generateTrg1Src2Instruction(cg, TR::InstOpCode::cmp8, node, cr0Reg, currentAddressReg, tempReg);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::beq, node, vectorLoopLabel, cr0Reg);
+
+   /*
+    * We are not on a 16-byte boundary, so we cannot directly load the first 16 bytes of the string
+    * for comparison. Instead, we load the 16 byte vector starting from the 16-byte aligned section
+    * containing the start of the string. Since we have 16-byte natural alignment, this can't cause
+    * an unexpected page fault.
+    */
+   generateTrg1MemInstruction(cg, TR::InstOpCode::lvx, node, searchVectorReg, TR::MemoryReference::createWithIndexReg(cg, zeroReg, currentAddressReg, 16));
+
+   /*
+    * However, before we can run any comparisons on the loaded vector, we must ensure that the extra
+    * garbage read before the start of the string can't match the target character. To do this, we
+    * shift the loaded vector backwards by n bytes shifting in the bitwise complement of the target
+    * character.
+    */
+   if (comp->target().cpu.isLittleEndian())
+      {
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::lvsr, node, permuteVectorReg, zeroReg, currentAddressReg); // Only the low 4 bits in currentAddressReg are used so it acts as a value between 0-15.
+      generateTrg1Src3Instruction(cg, TR::InstOpCode::vperm, node, searchVectorReg, targetVectorNotReg, searchVectorReg, permuteVectorReg);
+      }
+   else
+      {
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::lvsl, node, permuteVectorReg, zeroReg, currentAddressReg); // Only the low 4 bits in currentAddressReg are used so it acts as a value between 0-15.
+      generateTrg1Src3Instruction(cg, TR::InstOpCode::vperm, node, searchVectorReg, searchVectorReg, targetVectorNotReg, permuteVectorReg);
+      }
+
+   /*
+    * Now our vector is ready for comparison: no garbage can match the target value and the start of
+    * the vector is now aligned to the start of the string.
+    * This jumps to foundFirstVectorLabel to check the rest of the substring. If it doesn't match, it jumps back to firstContinueLabel.
```

Comment:
if it doesn't match, it falls through to firstContinueLabel, instead of jumping back to.

Reply:
My intent for this was to say if the branch was taken and the rest of the substring isn't found, it will jump back to `firstContinueLabel`. I edited the comment to be more clear.